src/java/arjdbc/postgresql/PostgreSQLResult.java

```diff
@@ -0,0 +1,173 @@
+package arjdbc.postgresql;
+
+import arjdbc.jdbc.JdbcResult;
+import arjdbc.jdbc.types.AbstractType;
+import arjdbc.jdbc.types.StringWithDefaultEncodingType;
+import arjdbc.postgresql.types.BitStringType;
+import arjdbc.postgresql.types.ObjectType;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyHash;
+import org.jruby.RubyString;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.Block;
+import org.jruby.runtime.ObjectAllocator;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import org.postgresql.core.Field;
+import org.postgresql.jdbc.PgResultSetMetaData;
+import org.postgresql.jdbc.PgResultSetMetaDataWrapper; // This is a hack unfortunately to get around method scoping
+
+
+/*
+ * This class mimics the PG:Result class enough to get by
+ */
+public class PostgreSQLResult extends JdbcResult {
+
+    // These are needed when generating an AR::Result
+    private IRubyObject adapter;
+    private PgResultSetMetaData resultSetMetaData;
+
+    /********* JRuby compat methods ***********/
+
+    private static RubyClass rubyClass;
```

Comment:
I wasn't quite sure what all needed to happen to make these `IRubyObject`s and to be able to include the `Enumerable` module in it. So if there is a better way/less code way to handle it I'm all ears.